.changeset/witty-walls-invent.md

```diff
@@ -0,0 +1,38 @@
+---
+"@aws-amplify/ui-react": patch
+"@aws-amplify/ui": patch
+---
+
+Adding support for token style props in React components and moving `createTheme` inside the AmplifyProvider so users don't have to call it directly in React. 
+
+```jsx
+const theme: Theme = {
+  name: 'my-theme',
+  tokens: {
+    //...
+  }
+}
+
+const App = () => {
+  return (
+    <AmplifyProvider theme={theme}>
+
+    </AmplifyProvider>
+  )
+}
+```
+
+Then using theme tokens in a style prop:
+
+```jsx
+import { useTheme, Text } from '@aws-amplify/ui-react';
+
+const MyComponent = () => {
+  const { tokens } = useTheme();
```

Comment:
In the PR video, I didn't see `tokens` referenced. So was that a static reference (e.g. `import tokens from "./my-style-tokens"`) or did I miss it?

Reply:
Good point, this was an old video before I did the provider change so I was using the object from `createTheme`. Recorded a new video:

https://user-images.githubusercontent.com/321279/141178411-bb3cd81f-2740-4b38-9955-74289c7cb4ca.mp4